### ISSUE TITLE
App crash on tapping `ADD SHIPPING ADDRESS`

### ISSUE THREAD

**Issue by pachlava** [woocommerce/woocommerce-android#5786]
**Describe the bug**
App crashes as soon as `ADD SHIPPING ADDRESS` is tapped in the order screen:

https://user-images.githubusercontent.com/73365754/152534759-3ff64abc-aac8-4a91-ba28-b374e1d2b48d.mov

**To Reproduce**
Steps to reproduce the behaviour:
1. Open an order with not yet added shipping address
2. Tap `ADD SHIPPING ADDRESS` ~> App crashes.

**Expected behaviour**
Shipping address form is shown

**Isolating the problem (mark completed items with an [x]):**
- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Mobile Environment**
Please include:
- Device: Samsung Galaxy A31
- Android version: Android 10
- WooCommerce Android version: 8.4-rc-1

**WordPress Environment**
<details>

```

### WordPress Environment ###

WordPress address (URL): https://automatticwidgets.wpcomstaging.com
Site address (URL): https://automatticwidgets.wpcomstaging.com
WC Version: 6.1.1
REST API Version: ✔ 6.1.1
WC Blocks Version: ✔ 6.5.2
Action Scheduler Version: ✔ 3.4.0
WC Admin Version: ✔ 3.1.0
Log Directory Writable: ✔
WP Version: 5.9
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: ✔

### Server Environment ###

Server Info: nginx
PHP Version: 7.4.27
PHP Post Max Size: 2 GB
PHP Time Limit: 1200
PHP Max Input Vars: 6144
cURL Version: 7.80.0
OpenSSL/1.1.0l

SUHOSIN Installed: –
MySQL Version: 5.5.5-10.3.21-MariaDB-log
Max Upload Size: 2 GB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 5.5.2
WC Database Prefix: wp_
Total Database Size: 3.94MB
Database Data Size: 2.37MB
Database Index Size: 1.57MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.05MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.02MB + Index: 0.13MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_failed_jobs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailchimp_carts: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_options: Data: 1.08MB + Index: 0.08MB + Engine InnoDB
wp_postmeta: Data: 0.14MB + Index: 0.13MB + Engine InnoDB
wp_posts: Data: 0.17MB + Index: 0.06MB + Engine InnoDB
wp_queue: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.05MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.06MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

amp_validated_url: 1
attachment: 21
jetpack_migration: 2
jp_img_sitemap: 6
jp_sitemap: 6
jp_sitemap_master: 6
nav_menu_item: 2
page: 6
post: 3
product: 9
product_variation: 7
revision: 3
shop_order: 24
shop_order_refund: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (10) ###

Akismet Anti-Spam: by Automattic – 4.2.2
AMP: by AMP Project Contributors – 2.2.1
Jetpack: by Automattic – 10.7-a.1
Storefront Powerpack: by WooCommerce – 2.0.0
WooCommerce Admin: by WooCommerce – 3.1.0
WooCommerce Stripe Gateway: by WooCommerce – 6.1.0
WooCommerce Product Add-ons: by WooCommerce – 4.5.0
WooCommerce Shipping & Tax: by WooCommerce – 1.25.22
WooCommerce Shipment Tracking: by WooCommerce – 1.6.14 (update to version 1.6.31 is available)
WooCommerce: by Automattic – 6.1.1

### Inactive Plugins (20) ###

Classic Editor: by WordPress Contributors – 1.6.2
CoBlocks: by GoDaddy – 2.19.1
Facebook for WooCommerce: by Facebook – 2.6.9
Gutenberg: by Gutenberg Team – 12.5.1
Klarna Checkout for WooCommerce: by Krokedil – 2.6.4
Klarna Payments for WooCommerce: by krokedil
klarna
automattic – 2.8.1

Layout Grid: by Automattic – 1.8
Mailchimp for WooCommerce: by Mailchimp – 2.5.4
Page Optimize: by Automattic – 0.5.2
TaxJar - Sales Tax Automation for WooCommerce: by TaxJar – 4.1.0
WooCommerce Australia Post Shipping: by WooCommerce – 2.4.32
WooCommerce Canada Post Shipping: by WooCommerce – 2.5.27
WooCommerce Eway Payment Gateway: by WooCommerce – 3.4.1
WooCommerce PayFast Gateway: by WooCommerce – 1.4.20
WooCommerce PayPal Checkout Gateway: by WooCommerce – 2.1.3
WooCommerce Royal Mail: by WooCommerce – 2.5.41
WooCommerce Square: by WooCommerce – 2.8.0
WooCommerce UPS Shipping: by WooCommerce – 3.2.37
WooCommerce USPS Shipping: by WooCommerce – 4.4.69
WordPress.com Editing Toolkit: by Automattic – 3.26189

### Dropin Plugins (2) ###

advanced-cache.php: advanced-cache.php
object-cache.php: Memcached

### Must Use Plugins (1) ###

WP.com Site Helper: by  –

### Settings ###

API Enabled: ✔
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: ✔

### WC Pages ###

Shop base: #20 - /shop/
Cart: #21 - /cart/
Checkout: #22 - /checkout/
My account: #23 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 3.9.1
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Action Scheduler ###

Complete: 32
Oldest: 2022-01-10 13:10:46 +0000
Newest: 2022-02-04 12:58:18 +0000

Failed: 1
Oldest: 2019-06-14 16:47:54 +0000
Newest: 2019-06-14 16:47:54 +0000

Pending: 2
Oldest: 2022-02-04 13:17:34 +0000
Newest: 2022-02-04 13:58:18 +0000


### Status report information ###

Generated at: 2022-02-04 13:18:06 +00:00

```

</details>

**Comment by shiki** [comment]
Reported in p5T066-2WL-p2#comment-11137

Stack trace:

```
2022-02-03 17:35:02.430 7811-7811/com.woocommerce.android.prealpha E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.woocommerce.android.prealpha, PID: 7811
    java.util.NoSuchElementException: Key SHIPPING is missing in the map.
        at kotlin.collections.MapsKt__MapWithDefaultKt.getOrImplicitDefaultNullable(MapWithDefault.kt:24)
        at kotlin.collections.MapsKt__MapsKt.getValue(Maps.kt:346)
        at com.woocommerce.android.ui.orders.details.editing.address.AddressViewModel.statesAvailableFor(AddressViewModel.kt:32)
        at com.woocommerce.android.ui.orders.details.editing.address.AddressViewModel.hasStatesFor(AddressViewModel.kt:69)
        at com.woocommerce.android.ui.orders.details.editing.address.BaseAddressEditingFragment.shouldShowStateSpinner(BaseAddressEditingFragment.kt:133)
        at com.woocommerce.android.ui.orders.details.editing.address.BaseAddressEditingFragment.updateStateViews(BaseAddressEditingFragment.kt:141)
        at com.woocommerce.android.ui.orders.details.editing.address.BaseAddressEditingFragment.onViewCreated(BaseAddressEditingFragment.kt:101)
        at androidx.fragment.app.Fragment.performViewCreated(Fragment.java:3019)
        at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:551)
        at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:261)
        at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1840)
        at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1764)
        at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1701)
        at androidx.fragment.app.FragmentManager$4.run(FragmentManager.java:488)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
        at android.app.ActivityThread.main(ActivityThread.java:7842)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
```

**Comment by wzieba** [comment]
Thank you! It's the same us #5744 and resolved in #5763
